examples/xds-management-server/cmd/controller.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"time"
+
+	agonescluster "quilkin.dev/xds-management-server/pkg/cluster/agones"
+	k8sfilterchain "quilkin.dev/xds-management-server/pkg/filterchain/k8s"
+	"quilkin.dev/xds-management-server/pkg/k8s"
+
+	"github.com/alecthomas/kong"
+	log "github.com/sirupsen/logrus"
+
+	"quilkin.dev/xds-management-server/pkg/server"
+	"quilkin.dev/xds-management-server/pkg/snapshot"
+)
+
+type flags struct {
+	KubeHost                string        `name:"kube-host" env:"KUBE_HOST" help:"The host the kubernetes API server runs on or 'cluster' if running in cluster" default:"cluster"`
+	KubeConfigPath          string        `name:"kube-config-path" env:"KUBE_CONFIG_PATH" help:"Path to local kube config"`
+	Port                    int16         `name:"int16" help:"Server listening port." default:"18000"`
+	ProxyNamespace          string        `name:"proxy-namespace" help:"Namespace under which the proxies run." default:"quilkin"`
+	GameServersNamespace    string        `name:"game-server-namespace" help:"Namespace under which the game-servers run." default:"gameservers"`
+	GameServersPollInterval time.Duration `name:"game-server-poll-interval" help:"How long to wait in-between checking for game-server updates." default:"1s"`
+	ProxyPollInterval       time.Duration `name:"proxy-interval" help:"How long to wait in-between checking for proxy updates." default:"1s"`
+}
+
+func main() {
+	var flags flags
+	kong.Parse(&flags)
+
+	logger := &log.Logger{}
+	logger.SetOutput(os.Stdout)
+	logger.SetLevel(log.DebugLevel)
+	logger.SetFormatter(&log.JSONFormatter{})
+
+	ctx, shutdown := context.WithCancel(context.Background())
+	defer shutdown()
+
+	k8sConfig, err := k8s.GetK8sConfig(logger, flags.KubeHost, flags.KubeConfigPath)
```

Comment:
Fun question for this: Does this mean there will be one xDS server per Kubernetes cluster?

An interesting assumption 😁 (lots of pros and cons I'm sure) - we should probably make a note of this in the README though.

Not got any strong opinions, especially as this is an example, but just highlighting it so we make a note of it somewhere in some documentation.

Reply:
Its not an assumption, rather this is an mvp and what I _think_ we need to test things out on our end. If it turns out we do need multiple clusters I'll add it but otherwise I'll circle back to it eventually or PR welcome if someone else wants to. We can create an issue to track it if we want

Reply:
Makes perfect sense 👍🏻 just want to make sure we cover this in a README.